### PR TITLE
plan: fix sort plan explain

### PR DESCRIPTION
**Pull request by ilovesoup**
Re-push fixing sort plan explain.
Still working on tests.

### CODE REVIEW

**Comment by shenli** [comment]
Can we pass tidb-test now?

File: plan/physical_plan_builder.go [hunk 1]
```diff
@@ -230,6 +230,9 @@ func (p *DataSource) convert2PhysicalPlan(prop *requiredProperty) (*physicalPlan
 			Columns:     p.Columns,
 			TableAsName: p.TableAsName,
 		}
+		memTable.tp = "MemTableScan"
```
Comment (shenli) [inline plan/physical_plan_builder.go]: Add a test case for this.

Comment (ilovesoup) [inline plan/physical_plan_builder.go]: done

**Comment by shenli** [comment]
LGTM
Please fix the conflicts.
@hanfei1991 PTAL

**Pull request merged**